### PR TITLE
Add WSL distros to profiles by default

### LINKED ISSUE #441: Does not detect wsl distro
* Your Windows build number:  10.0.18890.1000

* What you're doing and what's happening: Open windows terminal and press the down arrow, it shows cmd and PowerShell but not wsl, I have wsl debian installed.

* What's wrong / what should be happening instead: It should let me select wsl debian

### LINKED ISSUE #675: Auto-add a 'WSL' profile, linking to System32\wsl.exe
One of the features demo'd on the 'Introducing Windows Terminal' video was an automatic detection for WSL distributions. However, it's not yet been implemented.

I'm putting together a PR to fix this, but thus far, only for the default WSL distro (located by execution alias at `%WINDIR%\System32\wsl.exe`).

If anyone has a way to get a list of all possible linux distros installed locally, could they let me know so that I can add that as a feature too?

### PR DESCRIPTION
**Pull request by haydennyyy**
This closes #675

This will make sure that the user automatically gets a 'WSL' option in the new tab flyout.

EDIT: This also closes #441 thanks to @JBanks' code

### CODE REVIEW

File: src/cascadia/TerminalApp/CascadiaSettings.cpp [hunk 1]
```diff
@@ -26,72 +26,72 @@ CascadiaSettings::~CascadiaSettings()
...
+									  RGB(253, 246, 227),
+									  RGB(7, 54,  66) };
+	auto& solarizedDarkTable = solarizedDarkScheme.GetTable();
+	auto solarizedDarkSpan = gsl::span<COLORREF>(&solarizedDarkTable[0], gsl::narrow<ptrdiff_t>(COLOR_TABLE_SIZE));
+	solarizedDarkTable[0] = RGB(7, 54, 66);
+	solarizedDarkTable[1] = RGB(211, 1, 2);
+	solarizedDarkTable[2] = RGB(133, 153, 0);
+	solarizedDarkTable[3] = RGB(181, 137, 0);
+	solarizedDarkTable[4] = RGB(38, 139, 210);
+	solarizedDarkTable[5] = RGB(211, 54, 130);
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/CascadiaSettings.cpp]: Also, please back out unchanged portions (I bet this'll mostly go away after tabs switched back to spaces)

File: src/cascadia/TerminalApp/CascadiaSettings.cpp [hunk 2]
```diff
@@ -120,38 +120,60 @@ void CascadiaSettings::_CreateDefaultSchemes()
...
+	_profiles.emplace_back(defaultProfile);
+	_profiles.emplace_back(powershellProfile);
+
+	// If the user has installed the WSL, we add it as a default.
+	// We *don't* add it if the user hasn't installed WSL.
+	// WSL's default distro execution alias is "%WINDIR%\System32\wsl.exe".
+	std::filesystem::path wslCmdline{};
+	if (_IsWSLInstalled(L"%WINDIR%", wslCmdline))
+	{
+		std::wstring ubuntuCmdline = L"wsl.exe";
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/CascadiaSettings.cpp]: Unused variable

**Comment by haydennyyy** [comment]
> Please use spaces instead of tabs

Is there a quick way to do that in VS?

**Comment by carlos-zamora** [comment]
[https://stackoverflow.com/questions/14167033/visual-studio-replace-tab-with-4-spaces](https://stackoverflow.com/questions/14167033/visual-studio-replace-tab-with-4-spaces)

(you have no idea how many times this has saved me in code reviews haha)

---
In reply to: [491409840](https://github.com/microsoft/Terminal/pull/679#issuecomment-491409840) [](ancestors = 491409840)

**Comment by haydennyyy** [comment]
@carlos-zamora There, CI passed and tabs are spaces. :)

**Comment by binarycrusader** [comment]
> > Please use spaces instead of tabs
> 
> Is there a quick way to do that in VS? @carlos-zamora

There's also a handy tool in the Visual Studio Power Tools pack:

https://marketplace.visualstudio.com/items?itemName=VisualStudioPlatformTeam.ProductivityPowerPack2017

"Fix Mixed Tabs"

It automatically detects mixed indentation types on file open and asks if you want to fix them.

**Comment by haydennyyy** [comment]
Okay, so. There's a bit of an issue. While the code now compiles locally (I can't push the fix because I've also retargeted for the v142 build tools) WSL detection has broken...

**Comment by haydennyyy** [comment]
Duplicate of #695 (which does it better imo). Closing

**Pull request closed without merging**